apps/laboratory/tests/shared/pages/ModalPage.ts

```diff
@@ -446,13 +446,22 @@ export class ModalPage {
     await this.page.getByTestId('w3m-wallet-guide-get-started').click()
   }
 
-  async promptSiwe() {
-    const siweSign = this.page.getByTestId('w3m-connecting-siwe-sign')
-    await expect(siweSign, 'Siwe prompt sign button should be visible').toBeVisible({
-      timeout: 10_000
-    })
-    await expect(siweSign, 'Siwe prompt sign button should be enabled').toBeEnabled()
-    await siweSign.click()
+  async promptSiwe(cancel = false) {
+    if (cancel) {
+      const siweCancel = this.page.getByTestId('w3m-connecting-siwe-cancel')
+      await expect(siweCancel, 'Siwe prompt cancel button should be visible').toBeVisible({
+        timeout: 10_000
+      })
+      await expect(siweCancel, 'Siwe prompt cancel button should be enabled').toBeEnabled()
+      await siweCancel.click()
+    } else {
+      const siweSign = this.page.getByTestId('w3m-connecting-siwe-sign')
+      await expect(siweSign, 'Siwe prompt sign button should be visible').toBeVisible({
+        timeout: 10_000
+      })
+      await expect(siweSign, 'Siwe prompt sign button should be enabled').toBeEnabled()
+      await siweSign.click()
+    }
```

Comment:
nit: you could just select the element based on `cancel` and avoid repeating the code